src/main/scala/it/pagopa/interop/attributeregistryprocess/api/impl/AttributeRegistryApiServiceImpl.scala

```diff
@@ -140,40 +141,35 @@ final case class AttributeRegistryApiServiceImpl(
       } yield ()
 
       onComplete(result) {
-        loadCertifiedAttributesResponse[Unit](operationLabel)(_ => loadCertifiedAttributes200)
+        loadCertifiedAttributesResponse[Unit](operationLabel)(_ => loadCertifiedAttributes204)
       }
     }
 
-  def addNewAttributes(
-    attributeSeed: Seq[AttributeSeed]
-  )(implicit contexts: Seq[(String, String)]): Future[Set[Attribute]] = {
-
-    case class DeltaAttributes(attributes: Set[Attribute], seeds: Set[AttributeSeed]) {
-      def addAttribute(attr: Attribute): DeltaAttributes = copy(attributes = attributes + attr)
-
-      def addSeed(seed: AttributeSeed): DeltaAttributes = copy(seeds = seeds + seed)
-    }
+  private def addNewAttributes(
+    attributesSeeds: Seq[AttributeSeed]
+  )(implicit contexts: Seq[(String, String)]): Future[Unit] = {
 
     // calculating the delta of attributes
-    def delta(attrs: List[Attribute]): DeltaAttributes =
-      attributeSeed.foldLeft[DeltaAttributes](DeltaAttributes(Set.empty, Set.empty))((delta, seed) =>
+    def delta(attrs: Seq[PersistentAttribute]): Set[AttributeSeed] =
+      attributesSeeds.foldLeft[Set[AttributeSeed]](Set.empty)((delta, seed) =>
         attrs
-          .find(persisted => seed.name.equalsIgnoreCase(persisted.name))
-          .fold(delta.addSeed(seed))(delta.addAttribute)
+          .find(persisted => {
+            seed.origin.contains(persisted.origin.getOrElse("")) && seed.code.contains(persisted.code.getOrElse(""))
+          })
+          .fold(delta + seed)(_ => Set.empty)
       )
 
-    // for all the not existing attributes, execute the command to persist them through event sourcing
-    for {
-      attributesfromRM <- getAll(50)(readModelService.find[PersistentAttribute]("attributes", Filters.empty(), _, _))
-      deltaAttributes = delta(attributesfromRM.map(_.toApi).toList)
-      newlyCreatedAttributes <- Future.traverse(deltaAttributes.seeds)(attributeSeed =>
+    def getAndProcessAttributes(offset: Int, limit: Int): Future[Seq[PersistentAttribute]] = for {
+      attributesFromRM <- readModelService.find[PersistentAttribute]("attributes", Filters.empty(), offset, limit)
+      _                <- Future.traverse(delta(attributesFromRM))(attributeSeed => // create all new attributes
         attributeRegistryManagementService.createAttribute(attributeSeed.toClient)
       )
-    } yield deltaAttributes.attributes ++ newlyCreatedAttributes.map(_.toApi)
+    } yield attributesFromRM
 
+    getAndProcess(50)(getAndProcessAttributes).flatMap(_ => Future.unit)
```

Comment:
I realized now that my suggestion was not correct, sorry.

All existing attributes need to be retrieved at once, otherwise we will check the existence of an attribute that is probably in another page.

The previous solution was correct (but you can still avoid the `DeltaAttributes` class)